testsuite/MDAnalysisTests/analysis/test_contacts.py

```diff
@@ -181,6 +184,51 @@ def _run_Contacts(self, universe, start=None, stop=None, step=None, **kwargs):
             radius=6.0,
             **kwargs).run(start=start, stop=stop, step=step)
 
+    @pytest.mark.parametrize("seltxt", [sel_acidic, sel_basic])
+    def test_select_valid_types(self, universe, seltxt):
+        """Test if Contacts._get_atomgroup() can take both string and AtomGroup
+         as selections.
+        """
+        ag = universe.select_atoms(seltxt)
+
+        ag_from_string = contacts.Contacts._get_atomgroup(universe, seltxt)
+        ag_from_ag = contacts.Contacts._get_atomgroup(universe, ag)
+
+        assert_equal(ag_from_string, ag_from_ag)
```

Comment:
atomgroup comparison already calls np.array_equal in the background, doing a standard assert is orders of magnitude faster here (seconds to microseconds).
```suggestion
        assert ag_from_string == ag_from_ag
```